src/Compilers/CSharp/Portable/Utilities/InterceptableLocation.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+[Experimental(RoslynExperiments.Interceptors, UrlFormat = RoslynExperiments.Interceptors_Url)]
+public abstract class InterceptableLocation
+{
+    private protected InterceptableLocation() { }
+
+    /// <summary>
+    /// The version of the location encoding. Used as an argument to 'InterceptsLocationAttribute'.
+    /// </summary>
+    public abstract int Version { get; }
+
+    /// <summary>
+    /// Opaque data which references a call when used as an argument to 'InterceptsLocationAttribute'.
+    /// The value does not require escaping, i.e. it is valid in a string literal when wrapped in " (double-quote) characters.
+    /// </summary>
+    public abstract string Data { get; }
+
+    /// <summary>
+    /// Gets a human-readable representation of the location, suitable for including in comments in generated code.
+    /// </summary>
+    public abstract string GetDisplayLocation();
+}
+
+#pragma warning disable RSEXPERIMENTAL002 // internal usage of experimental API
+/// <summary>
+/// Version 1 of the InterceptableLocation encoding.
+/// </summary>
+internal sealed class InterceptableLocation1 : InterceptableLocation
```

Comment:
```suggestion
internal sealed class InterceptableLocationVersion1 : InterceptableLocation
```

this way we could have dinstinct separate types per version.